src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7193,7 +7193,7 @@ public override BoundNode VisitTypeExpression(BoundTypeExpression node)
                 VisitTypeExpression(node.BoundContainingTypeOpt);
             }
 
-            SetNotNullResult(node);
+            SetResult(node, node.TypeWithAnnotations.ToTypeWithState(), node.TypeWithAnnotations);
```

Comment:
>SetResult(node, node.TypeWithAnnotations.ToTypeWithState(), node.TypeWithAnnotations); [](start = 12, length = 86)

Not all type references have corresponding BoundTypeExpression nodes. For example, those used in declarations don't. If the goal of adjusting this line is to adjust SemanticModel's behavior, it feels like not all scenarios are going to be handled.

Reply:
Scenarios for types in member declarations will not go through this code path. How are they handled?

---
In reply to: [366367464](https://github.com/dotnet/roslyn/pull/40950#discussion_r366367464) [](ancestors = 366367464)